src/debug.c

```diff
@@ -2184,6 +2183,28 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
     bugReportEnd(1, sig);
 }
 
+void setupSigSegvHandler(void) {
+    /* Set signal handler with error checking attribute. re-lock within the same thread will error. */
+    pthread_mutexattr_init(&signal_handler_lock_attr);
+    pthread_mutexattr_settype(&signal_handler_lock_attr, PTHREAD_MUTEX_ERRORCHECK);
+    pthread_mutex_init(&signal_handler_lock, &signal_handler_lock_attr);
```

Comment:
setupSignalHandlers can be called repeatedly at runtime (see `updateSighandlerEnabled`).
i prefer you'll move the `sigaction` calls below back to server.c (smaller diff).
and maybe add an `initDebug` to be called from `initServer`.

the other alternative is to use another trick to avoid re-initing these.